File: baekjoon/python/2018.py
N = int(input())
sum = 1
right =1
left =1
# numbers = [i for i in range(1, N+1)]
tot = 0
while (right <= N and left <= right):
    if (sum > N):
        sum -= left
        # sum -= numbers[left]
        left += 1
    elif (sum == N):
        tot += 1
        sum -= left
        left += 1
    else:
        right += 1
        # print(right)
        sum += right

print(tot)

File: baekjoon/c++/2018.cpp


#include <iostream>
using namespace std;

int main() {
    int N;
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    cin >> N;
    int sum = 1;
    int right = 1;
    int left = 1;
    int tot = 0;
    while (right <= N && left <= right) {
        if (sum > N) {
            sum -= left;
            left += 1;
        } else if (sum == N) {
            tot += 1;
            sum -= left;
            left += 1;
        } else {
            right += 1;
            sum += right;
        }
    }
    cout << tot << endl;
    return 0;
}

